compiler/parser/RdrHsSyn.hs

```diff
@@ -2688,7 +2706,7 @@ mkChoiceDecls templateLoc conName binds (CombinedChoiceData controllers observer
         controllerSig = mkFunTy templateType (mkFunTy choiceType partiesType)
         controllerDef = mkLambda controllerDefArgs controllers (Just (extendLetBindings binds (dummyBinds controllerDefArgs)))
         controllerDefArgs = [this, if flexible then arg else WildPat noExt]
-        observersSig = mkOptional (mkFunTy templateType (mkFunTy choiceType partiesType))
```

Comment:
before:

```
_choice_... : (..., ..., ..., Optional Template -> Choice -> [Party])
```
after:
```
_choice_... : (..., ..., ..., Optional (Template -> Choice -> [Party]))
```

Reply:
this only affects pretty printing right? iirc doesn’t actually care about the parens being there or not.

Reply:
just to be clear: all in favor of fixing it even if it does only affect pretty printing

Reply:
yup! it's only because I noticed when updating `InterfaceDesugared.daml` and got weird type errors. Thankfully this seemed like the only such instance.